eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -370,6 +431,595 @@ def test_update_user_in_tenant_user_not_found(self, param: str, value: str) -> N
+        - `get_edxapp_user`
+        - `create_enrollment`
+        - `check_edxapp_account_conflicts`
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 200.
+        - The enrollment is created successfully in the tenant with the provided data.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "mode": self.mode,
+        }
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data["username"], user_data["username"])
+        self.assertEqual(response_data["mode"], enrollment_data["mode"])
+        self.assertEqual(response_data["course_id"], enrollment_data["course_id"])
+        self.assertTrue(response_data["is_active"])
+        self.assertIn("created", response_data)
+
+    @ddt.data(
+        ("mode", {"mode": ["This field is required."]}),
+        ("course_id", {"non_field_errors": ["You have to provide a course_id or bundle_id"]}),
+        ("username", {"non_field_errors": ["Email or username needed"]}),
+    )
+    @ddt.unpack
+    def test_create_enrollment_missing_required_fields(self, param: str, error: list | dict) -> None:
+        """
+        Test creating a enrollment with missing required fields.
+
+        Open edX definitions tested:
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains a message about the missing field.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            "username": user_data["username"],
+            "mode": self.mode,
+            "course_id": self.course_id,
+        }
+        enrollment_data.pop(param)
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, error)
+
+    @ddt.data("email", "username")
+    def test_force_create_enrollment_success(self, param: str) -> None:
+        """
+        Test force creating a enrollment with a valid user, course and mode in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `create_enrollment`
+        - `check_edxapp_account_conflicts`
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 200.
+        - The enrollment is created successfully in the tenant with the provided data.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
```

Comment:
Do we have to create the users each time? Can we make it part of the setup?

Reply:
The problem is that we need the user's data in each test. I'm not sure how we would access it afterwards. Do you have any idea?

Reply:
I think I know how to do it, I'll do some tests.

Reply:
Done: https://github.com/eduNEXT/eox-core/commit/bd1d3c1ef1c7c0aea3864dbe491df456a0b0b43f